MIG-1418: MTC 1.8.0 release notes (#65437)
openshift/openshift-docs: modules/migration-mtc-release-notes-1-8.adoc

(link:https://bugzilla.redhat.com/show_bug.cgi?id=2233097[*BZ#2233097*])

.PodVolumeRestore remain In Progress keeping the migration stuck at Stage Restore

In previous releases, on performing an indirect migration, the migration became stuck at the `Stage Restore` step, waiting for the `podvolumerestore` to be completed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2233868[*BZ#2233868*])

.Migrated application unable to pull image from internal registry on target cluster

In previous releases, on migrating an application to the target cluster, the migrated application failed to pull the image from the internal image registry resulting in an `application failure`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2233103[*BZ#2233103*])

.Migration failing on Azure due to authorization issue

In previous releases, on an Azure cluster, when backing up to Azure storage, the migration failed at the `Backup` stage. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2238974[*BZ#2238974*])

[id="known-issues-1-8_{context}"]
== Known issues

This release has the following known issues:

.Old Restic pods are not getting removed on upgrading MTC 1.7.x -> 1.8.x

> In this release, on upgrading the MTC operator from 1.7.x to 1.8.x, the old Restic pods are not being removed. Hence, after the upgrade, both Restic and node-agent pods are visible in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2236829[*BZ#2236829*])

Reviewer: ```suggestion
In this release, on upgrading the MTC operator from 1.7.x to 1.8.x, the old Restic pods are not being removed. Therefore after the upgrade, both Restic and node-agent pods are visible in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2236829[*BZ#2236829*])
```

> In this release, on upgrading the MTC operator from 1.7.x to 1.8.x, the old Restic pods are not being removed. Therefore after the upgrade, both Restic and node-agent pods are visible in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2236829[*BZ#2236829*])

Reviewer 2: ```suggestion
In this release, on upgrading the MTC Operator from 1.7.x to 1.8.x, the old Restic pods are not being removed. Therefore after the upgrade, both Restic and node-agent pods are visible in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2236829[*BZ#2236829*])
```

.Migrated builder pod fails to push to image registry

In this release, on migrating an application including a `BuildConfig` from a source to target cluster, builder pod results in `error`, failing to push the image to the image registry. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2234781[*BZ#2234781*])

.[UI] CA bundle file field is not properly cleared

In this release, after enabling `Require SSL verification` and adding content to the CA bundle file for an MCG NooBaa bucket in MigStorage, the connection fails as expected. However, when reverting these changes by removing the CA bundle content and clearing `Require SSL verification`, the connection still fails. The issue is only resolved by deleting and re-adding the repository. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2240052[*BZ#2240052*])

[id="technical-changes-1-8_{context}"]
== Technical changes

This release has the following technical changes:

> * Migration from {product-title} 3 to {product-title} 4 requires a legacy {mtc-full} ({mtc-short}) Operator and {mtc-short} 1.7.x.

Reviewer: ```suggestion
* Migration from {product-title} 3 to {product-title} 4 requires a traditional {mtc-full} ({mtc-short}) Operator and {mtc-short} 1.7.x.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#l

* Migration from {mtc-short} 1.7.x to {mtc-short} 1.8.x is not supported.
* You must use {mtc-short} 1.7.x to migrate anything with a source of {product-title} 4.9 or earlier.
** {mtc-short} 1.7.x must be used on both source and destination.

> * MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x may be used. However, but it must be the same MTC 1.Y.z on both source & destination.

Reviewer: ```suggestion
* MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x might be used. However, but it must be the same MTC 1.Y.z on both source & destination.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m

> * MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x might be used. However, but it must be the same MTC 1.Y.z on both source & destination.

Reviewer 2: ```suggestion
* MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x might be used. However, but it must be the same MTC 1.Y.z on both source and destination.
```

** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.8.x is unsupported.
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.7.x is unsupported.
** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.7.x is supported.

> ** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.8.x is supported

Reviewer 2: ```suggestion
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.8.x is supported.
```

* MTC 1.8.x by default installs OADP 1.2.x.

> * Upgrading from {mtc-short} 1.7.x to {mtc-short} 1.8.0, requires manually changing the OADP channel to 1.2. If this is not done, the upgrade of the operator will fail.

Reviewer: ```suggestion
* Upgrading from {mtc-short} 1.7.x to {mtc-short} 1.8.0, requires manually changing the OADP channel to 1.2. If this is not done, the upgrade of the operator fails.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice

> * Upgrading from {mtc-short} 1.7.x to {mtc-short} 1.8.0, requires manually changing the OADP channel to 1.2. If this is not done, the upgrade of the operator fails.

Reviewer 2: ```suggestion
* Upgrading from {mtc-short} 1.7.x to {mtc-short} 1.8.0, requires manually changing the OADP channel to 1.2. If this is not done, the upgrade of the Operator fails.
```
